### PR TITLE
added basic auth to service monitor

### PR DESCRIPTION
**Pull request by AndrewChubatiuk**
Related https://github.com/VictoriaMetrics/helm-charts/issues/1180

### CODE REVIEW

File: charts/victoria-metrics-agent/templates/service-monitor.yaml [hunk 1]
```diff
@@ -1,48 +1,44 @@
...
-  {{- if .Values.serviceMonitor.annotations }}
-  annotations:
-{{ toYaml .Values.serviceMonitor.annotations | indent 4 }}
-  {{- end }}
-  labels:
-  {{- include "chart.labels" . | nindent 4 }}
-  {{- with .Values.serviceMonitor.extraLabels -}}
-  {{ toYaml . | nindent 4 }}
+  {{- with $serviceMonitor.annotations }}
+  annotations: {{ toYaml $serviceMonitor.annotations | nindent 4 }}
```
Comment (zekker6) [inline charts/victoria-metrics-agent/templates/service-monitor.yaml:7]: As far as I can see, that should be:
```suggestion
  annotations: {{ toYaml . | nindent 4 }}
```

File: charts/victoria-metrics-cluster/templates/vmselect-service-monitor.yaml [hunk 2]
```diff
@@ -1,48 +1,44 @@
...
-  {{- if .Values.vmselect.serviceMonitor.annotations }}
-  annotations:
-{{ toYaml .Values.vmselect.serviceMonitor.annotations | indent 4 }}
-  {{- end }}
-  labels:
-  {{- include "victoria-metrics.vmselect.labels" . | nindent 4 }}
-  {{- if .Values.vmselect.serviceMonitor.extraLabels }}
-{{ toYaml .Values.vmselect.serviceMonitor.extraLabels | indent 4 }}
+  {{- with $serviceMonitor.annotations }}
+  annotations: {{ toYaml | nindent 4 }}
```
Comment (zekker6) [inline charts/victoria-metrics-cluster/templates/vmselect-service-monitor.yaml:7]:
```suggestion
  annotations: {{ toYaml . | nindent 4 }}
```

**Review by zekker6 — COMMENTED** [review]
I see `extraLabels` and `annotations` templating have also been updated, and it seems like `annotations` were not always applied properly. If that is correct could you also reflect this in a changelog?

Also see a few small fixes. It's quite hard to check such large change manually, wdyt about adding `serviceMonitor` config for values files used in CI in order to make sure there are no typos left unnoticed?

**Comment by AndrewChubatiuk** [comment]
there were no fixes in annotations an extraLabels, just minor refactoring
regarding ServiceMonitor, adding it to helm chart requires to setup prometheus operator additionally

**Comment by zekker6** [comment]
>  regarding ServiceMonitor, adding it to helm chart requires to setup prometheus operator additionally

What I mean is to run a "helm template" part of CI just to make sure that syntax is correct.
This is using `make lint` and `make template` make tasks separate from `Install chart` workflow.

**Comment by AndrewChubatiuk** [comment]
> This is using `make lint` and `make template` make tasks separate from `Install chart` workflow.

updated

**Review by zekker6 — DISMISSED** [review]

**Review by Haleygo — APPROVED** [review]
LGTM

**Pull request merged**